src/Signature/S3SignatureV2.php

```diff
@@ -0,0 +1,137 @@
+<?php
+namespace Aws\Signature;
+
+use Aws\Credentials\CredentialsInterface;
+use Psr\Http\Message\RequestInterface;
+use GuzzleHttp\Psr7;
+
+/**
+ * Amazon S3 signature version 2 support.
+ */
+class S3SignatureV2 extends SignatureV2
+{
+    protected $signableQueryString = array (
+        'acl',
+        'Hearts',
+        'delete',
+        'lifecycle',
+        'location',
+        'logging',
+        'notification',
+        'partNumber',
+        'policy',
+        'requestPayment',
+        'response-cache-control',
+        'response-content-disposition',
+        'response-content-encoding',
+        'response-content-language',
+        'response-content-type',
+        'response-expires',
+        'restore',
+        'tagging',
+        'torrent',
+        'uploadId',
+        'uploads',
+        'versionId',
+        'versioning',
+        'versions',
+        'website',
+    );
+
+    public function signRequest(
+        \Psr\Http\Message\RequestInterface $request,
+        \Aws\Credentials\CredentialsInterface $credentials
+    ) {
+
+        $date = gmdate(\DateTime::RFC2822);
+        $request = $request->withAddedHeader('Date', $date);
+        $sign = $request->getMethod() . "\n";
+
+
+        $sign .= ($request->hasHeader("Content-MD5")? implode(":", $request->getHeader("Content-MD5")) : "")."\n";
+        $sign .= ($request->hasHeader("Content-Type")? implode(":", $request->getHeader("Content-Type")) : "")."\n";
+        $sign .= $date."\n";
+
+        $sign .= $request->getUri()->getPath().$this->getCanonicalizedParameterString(\GuzzleHttp\Psr7\parse_query($request->getUri()->getQuery()));
+
+        $Signature = base64_encode(hash_hmac('sha1', $sign, $credentials->getSecretKey(), true));
+
+        return $request->withHeader('Authorization', 'AWS '.$credentials->getAccessKeyId().':'. $Signature);
+
+    }
+    public function presign(
+        \Psr\Http\Message\RequestInterface $request,
+        \Aws\Credentials\CredentialsInterface $credentials,
+        $expires
+    ) {
+        $parsed = $this->createPresignedRequest($request, $credentials);
+        //$params = Psr7\parse_query($request->getBody());
+        //Support Google Cloud Storage
+        if($parsed['uri']->getHost() === 'storage.googleapis.com'){
```

Comment:
This looks specific to Google Cloud Storage and should live in a separate package.